src/solver_py.cpp

```diff
@@ -49,18 +50,151 @@ void Solver_Py::extract_options(const po::variables_map &vm __attribute__((unuse
 
 }
 
-void Solver_Py::add_edge(const string &src_path __attribute__((unused)),
-                         const string &statement __attribute__((unused)),
-                         unsigned int idx __attribute__((unused)),
-                         unsigned int line_no __attribute__((unused))) {
+void Solver_Py::add_edge(const string &src_path,
+                         const string &statement,
+                         unsigned int idx,
+                         unsigned int line_no) {
+    using boost::filesystem::path;
+    using boost::filesystem::operator/;
+
```

Comment:
Mutex locking missing?

Reply:
It relates to the issue below, please have a look. It's not missing, it appears later because everything before it does not modify the graph, which has to be protected. :)